lib/ash/actions/read/read.ex

```diff
@@ -2623,6 +2623,24 @@ defmodule Ash.Actions.Read do
     }
   end
 
+  defp add_relationship_limit_to_loads(%Ash.Query{load: []} = query), do: query
```

Comment:
We should do this in `relationships.ex` when we build the related query, not here 😄